src/Collector/ClassNameCollector.php

```diff
@@ -33,9 +22,18 @@ public function satisfy(
         AstParserInterface $astParser
     ) {
         return preg_match(
-            '/'.$this->getRegexByConfiguration($configuration).'/i',
+            $this->getConfigurationRegex($configuration),
             $abstractClassReference->getClassName(),
             $collectorFactory
         );
     }
+
+    private function getConfigurationRegex(array $configuration)
```

Comment:
ByConfiguration just indicates the argument (because it's just an array)
`byConfiguration` means the input is `array $configuration`.

what do you mean by "that's not a regex"? `$configuration['regex'];` is and was a regex?